src/components/ShareMatchExperienceModal/index.tsx

```diff
@@ -94,12 +74,17 @@ const CreateMatchExperienceModal = ({ isOpen, onClose }: CreateMatchExperienceMo
         SoccerService.getVenues
     );
 
+    const { data: fetchedTeams = [] } = useQueryOnDefinedParam(
+        'teams',
+        selectedLeague && selectedDate
+            ? { leagueId: selectedLeague, season: calculateCurrentSeason(new Date(selectedDate)) }
+            : undefined,
+        ({ leagueId, season }) => SoccerService.getTeams({ leagueId, season })
+    );
+
     useEffect(() => {
-        if (selectedLeague !== undefined && selectedDate !== null) {
-            const season = handleYear(new Date(selectedDate));
-            SoccerService.getTeams({ leagueId: selectedLeague, season }).then((data) => setTeams(data));
-        }
-    }, [selectedLeague, selectedDate]);
+        setTeams(fetchedTeams);
```

Comment:
Why you need teams and also the fetchedTeams? they are always the same arent they?

Reply:
```suggestion
const queryClient = useQueryClient();

const resetTeams = () => {
    queryClient.setQueryData(['teams'], []); // Clears the existing teams immediately
};
```

you can clear also the fetched teams this way without making new state and `useEffect` to sync them. try it. I think it should work. if not rollback to this